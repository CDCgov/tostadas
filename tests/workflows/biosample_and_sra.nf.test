nextflow_workflow {

    name "Test Workflow BIOSAMPLE_AND_SRA"
    script "workflows/biosample_and_sra.nf"
    workflow "BIOSAMPLE_AND_SRA"

test("Should run BioSample and SRA submission from validated metadata and templates") {
        when {
            params {
                meta_path = "${projectDir}/assets/sample_metadata/rsv_test_metadata.xlsx"
                outdir = "${projectDir}/results"
                
                annotation = false
                submission = true
                sra = true
                biosample = true
                genbank = false
                dry_run = true
                species = 'rsv'

                submission_config = "${projectDir}/conf/submission_config.yaml"
                help = false
            }
        }

        then {
            assert workflow.success

        }
    }

}
