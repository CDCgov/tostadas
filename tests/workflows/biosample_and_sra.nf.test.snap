{
    "Should run BioSample and SRA submission from validated metadata and templates": {
        "content": [
            {
                "0": [
                    "validated_metadata_all_samples.tsv:md5,ca071cf1536ab43ae483d32521f9219e"
                ],
                "1": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,ca071cf1536ab43ae483d32521f9219e"
                        },
                        [
                            [
                                "submission.xml:md5,befc33a338d4b11ac75330ea532d51d6",
                                "submit.ready:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ],
                            "prep_submission.log:md5,50b85e0c2e48bd0f7ba621332580baaf",
                            "submission.log:md5,fefa479f846dbe2985e2fbc9d1777a37"
                        ]
                    ]
                ],
                "submission_batch_folder": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,ca071cf1536ab43ae483d32521f9219e"
                        },
                        [
                            [
                                "submission.xml:md5,befc33a338d4b11ac75330ea532d51d6",
                                "submit.ready:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ],
                            "prep_submission.log:md5,50b85e0c2e48bd0f7ba621332580baaf",
                            "submission.log:md5,fefa479f846dbe2985e2fbc9d1777a37"
                        ]
                    ]
                ],
                "validated_concatenated_tsv": [
                    "validated_metadata_all_samples.tsv:md5,ca071cf1536ab43ae483d32521f9219e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T16:33:45.985695997"
    }
}